src/vs/workbench/contrib/chat/browser/actions/chatClearActions.ts

```diff
@@ -180,29 +180,35 @@ export function registerNewChatActions() {
 			const widgetService = accessor.get(IChatWidgetService);
 			const dialogService = accessor.get(IDialogService);
 			const viewsService = accessor.get(IViewsService);
+			const agentService = accessor.get(IChatAgentService);
+
 			if (!(await this._handleCurrentEditingSession(editingSession, dialogService))) {
 				return;
 			}
+
+			let widget: IChatWidget | undefined;
 			if (isChatViewTitleActionContext(context)) {
 				// Is running in the Chat view title
-				announceChatCleared(accessibilitySignalService);
-				const widget = widgetService.getWidgetBySessionId(context.sessionId);
-				if (widget) {
-					await editingSession.stop(true);
-					widget.clear();
-					widget.attachmentModel.clear();
-					widget.focusInput();
-				}
+				widget = widgetService.getWidgetBySessionId(context.sessionId);
 			} else {
 				// Is running from f1 or keybinding
 				const chatView = await viewsService.openView(EditsViewId) as ChatViewPane;
-				const widget = chatView.widget;
+				widget = chatView.widget;
+			}
 
-				announceChatCleared(accessibilitySignalService);
+			announceChatCleared(accessibilitySignalService);
+			if (widget) {
 				await editingSession.stop(true);
 				widget.clear();
 				widget.attachmentModel.clear();
 				widget.focusInput();
+
+				agentService.toggleToolsAgentMode(typeof context?.agentMode === 'boolean' ? context.agentMode : undefined);
```

Comment:
Mostly looks good but can you add an interface for the type that will come from extensions, similar to `IChatViewOpenOptions`? Including the `isPartialQuery` option. Do you have to auto-submit the query in your scenario? And `context.agentMode` should only be checked when not `isChatViewTitleActionContext`.

Reply:
Yeah, this is fine with me.